atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/UnstableOrderedIterableTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+
+public class UnstableOrderedIterableTest {
+    @Test
+    public void orderingIsUnstable() {
+        // Strobes once in 1000000! times. We can live with that.
+        List<Integer> numbers = IntStream.range(0, 1000000).boxed().collect(Collectors.toList());
```

Comment:
nope nope nope. Nondeterministic tests are bad. In this case you have to create 1 million objects.

Instead, UnstableOrderedIterable should take as a field a Random, and you can instantiate that deterministically here, use the version of shuffle that takes a Random and guarantee it 100% of the time.

Reply:
with 2 elements

Reply:
Actually, you're doing shuffling here, instead you should have UnstableOrderedIterable return OrderedPermutations.



the test I'd like to see here is one that basically reads:

```java
Iterable<Integer> numbers = new UnstableOrderedIterable<>(ImmutableList.of(1, 2));
assertThat(ImmutableList.copyOf(numbers)).containsExactly(1, 2);
assertThat(ImmutableList.copyOf(numbers)).containsExactly(2, 1);
```